src/server/game/Handlers/SpellHandler.cpp

```diff
@@ -591,14 +591,14 @@ void WorldSession::HandleSelfResOpcode(WorldPacket & /*recvData*/)
 {
     TC_LOG_DEBUG("network", "WORLD: CMSG_SELF_RES");                  // empty opcode
 
-    if (_player->HasAuraType(SPELL_AURA_PREVENT_RESURRECTION))
-        return; // silent return, client should display error by itself and not send this opcode
-
-    if (uint32 spellId = _player->GetUInt32Value(PLAYER_SELF_RES_SPELL))
-    {
-        _player->CastSpell(_player, spellId);
-        _player->SetUInt32Value(PLAYER_SELF_RES_SPELL, 0);
-    }
+	if (uint32 spell = sSpellMgr->GetSpellInfo(_player->GetUInt32Value(PLAYER_SELF_RES_SPELL)))
```

Comment:
it seems some tab characters ended up here, could you replace them with spaces (the usual 4 spaces for 1 tab) ?